File: app/src/main/java/com/pkfcooperparry/template/data/model/db/visit/Element.java
package com.pkfcooperparry.template.data.model.db.visit;

import com.google.gson.annotations.Expose;
import com.google.gson.annotations.SerializedName;

import io.realm.RealmList;
import io.realm.RealmObject;
import io.realm.annotations.PrimaryKey;
import lombok.Getter;
import lombok.Setter;

public class Element extends RealmObject {
    @Getter
    @Setter
    @SerializedName("id")
    @Expose
    @PrimaryKey
    private long id;
    @Getter
    @Setter
    @SerializedName("elementType")
    @Expose
    private String elementType;
    @Getter
    @Setter
    @SerializedName("name")
    @Expose
    private String name;
    @Getter
    @Setter
    @SerializedName("enabled")
    @Expose
    private Boolean enabled = false;
    @Getter
    @Setter
    @SerializedName("hint")
    @Expose
    private String hint;
    @Getter
    @Setter
    @SerializedName("choices")
    @Expose
    private RealmList<Choice> choices = null;
    @Getter
    @Setter
    @SerializedName("defaultValue")
    @Expose
    private Value defaultValue;
    @Getter
    @Setter
    @SerializedName("enabledWhen")
    @Expose
    private Condition enabledWhen;
    @Getter
    @Setter
    @SerializedName("showWhen")
    @Expose
    private Condition showWhen;
    @Getter
    @Setter
    @SerializedName("moreInfoElement")
    @Expose
    private Element moreInfoElement;
    @Getter
    @Setter
    @SerializedName("autoSelectCheckboxes")
    @Expose
    private RealmList<Long> autoSelectCheckboxes = null;
    @Getter
    @Setter
    @SerializedName("updateGroup")
    @Expose
    private RealmList<Long> updateGroup;
    @Getter
    @Setter
    @SerializedName("autoSelectDropdown")
    @Expose
    private Long autoSelectDropdown;

    public Element() {
    }
}

File: app/src/main/java/com/pkfcooperparry/template/data/model/db/visit/Element.kt
package com.pkfcooperparry.template.data.model.db.visit

import com.google.gson.annotations.Expose
import com.google.gson.annotations.SerializedName

import io.realm.RealmList
import io.realm.RealmObject
import io.realm.annotations.PrimaryKey
import lombok.Getter
import lombok.Setter

open class Element : RealmObject() {

    @SerializedName("id")
    @Expose
    @PrimaryKey
    var id: Long = 0

    @SerializedName("elementType")
    @Expose
    var elementType: String? = null

    @SerializedName("name")
    @Expose
    var name: String? = null

    @SerializedName("enabled")
    @Expose
    var enabled: Boolean? = false

    @SerializedName("hint")
    @Expose
    var hint: String? = null

    @SerializedName("choices")
    @Expose
    var choices: RealmList<Choice>? = null

    @SerializedName("defaultValue")
    @Expose
    var defaultValue: Value? = null
    @SerializedName("enabledWhen")
    @Expose
    var enabledWhen: Condition? = null
    @SerializedName("showWhen")
    @Expose
    var showWhen: Condition? = null
    @SerializedName("moreInfoElement")
    @Expose
    var moreInfoElement: Element? = null
    @SerializedName("autoSelectCheckboxes")
    @Expose
    var autoSelectCheckboxes: RealmList<Long>? = null
    @SerializedName("updateGroup")
    @Expose
    var updateGroup: RealmList<Long>? = null
    @SerializedName("autoSelectDropdown")
    @Expose
    var autoSelectDropdown: Long? = null
}
